### PR TITLE
Put minid3 in src/lib

### PR DESCRIPTION
**Pull request by danvk**
See #275 

This will let other programs (like cycledash) depend on pileup again.

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/pileup.js/318)

### CODE REVIEW

**Comment by armish** [comment]
:+1: just two questions for clarification.


---

Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[package.json, line 28 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/318#-K16DavecayspLD7fHTu)** ([raw file](https://github.com/hammerlab/pileup.js/blob/e4cc8952859c9af0b5b8726ffb527859ba9f3f07/package.json#L28)):</sup>
is there a reason why you remove this task but directly call it from other scripts?


---

<sup>**[scripts/watch.sh, line 13 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/318#-K16EhFyZgxPVX41j5EQ)** ([raw file](https://github.com/hammerlab/pileup.js/blob/bf11eb4499bc99bdf49c09cdbc28c5cc65d62401/scripts/watch.sh#L13)):</sup>
don't you need to carry the minified d3 over to the dist, similar to the way you do in `build.sh`? I might be wrong, but if somebody calls `watch.sh` before `build.sh` that mini3d won't be there, right?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/318)

**Comment by danvk** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[package.json, line 28 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/318#-K16DavecayspLD7fHTu)** ([raw file](https://github.com/hammerlab/pileup.js/blob/e4cc8952859c9af0b5b8726ffb527859ba9f3f07/package.json#L28)):</sup>
Because it's not something I could ever imagine a user wanting to run directly. It's more of an intermediate step.


---

<sup>**[scripts/watch.sh, line 13 \[r1\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/318#-K16EhFyZgxPVX41j5EQ)** ([raw file](https://github.com/hammerlab/pileup.js/blob/bf11eb4499bc99bdf49c09cdbc28c5cc65d62401/scripts/watch.sh#L13)):</sup>
Interestingly enough, no! These `watchify` tasks only take inputs from `src/`, not from `dist/`. In the new world of babel, the individual files in `dist/main` are only needed for distribution. The babel transforms are done inside the browserify/watchify processes.

In other words, they pull in `src/lib/minid3.js`, not `dist/lib/minid3.js`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/318)
